cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -158,15 +158,18 @@ public void initialize(TwillContext context) {
 
     try {
       CommandLine cmdLine = parseArgs(context.getApplicationArguments());
+      ClassLoader classLoader = getClass().getClassLoader();
 
       // Loads configurations
       hConf = new Configuration();
       hConf.clear();
-      hConf.addResource(new File(cmdLine.getOptionValue(RunnableOptions.HADOOP_CONF_FILE)).toURI().toURL());
+      hConf.addResource(classLoader.getResource(cmdLine.getOptionValue(RunnableOptions.HADOOP_CONF_FILE)));
 
       UserGroupInformation.setConfiguration(hConf);
 
-      cConf = CConfiguration.create(new File(cmdLine.getOptionValue(RunnableOptions.CDAP_CONF_FILE)));
```

Comment:
Not really a major difference, since cConf and hConf were added as resources, they are available via classloader. It's cleaner to get it from the classloader.